src/main/java/de/tum/in/www1/artemis/domain/GradingInstruction.java

```diff
@@ -0,0 +1,155 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+/**
+ * A Structured Grading Instruction.
+ */
+@Entity
+@Table(name = "grading_instruction")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+public class GradingInstruction implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    private Exercise exercise;
+
+    @Column(name = "credits")
+    private double credits;
+
+    @Column(name = "grading_scale")
+    private String gradingScale;
```

Comment:
Is this nullable? Moreover, which of the attributes should be nullable, and which should not?

Reply:
I did not add such limitations yet. I think I need to discuss these design decisions with Stephan first.